### PR TITLE
Add docs for grouping conversations

### PR DESCRIPTION
**Pull request by saileshdev**
Node, sub-story, story distinction
PT-759

### CODE REVIEW

File: bot-builder/creating-bot.rst [hunk 1]
```diff
@@ -1,2 +1,80 @@
 Creating a Bot
 --------------
+
+Designing chatbot interactions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The designer can model the conversation flow based on the type of interactions between the user and a chatbot.
+These are segmented into structured and unstructured interactions. As the name suggests, the structured type is more
+about the logical flow of information including menus, choices and forms into account.
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: about the logical flow of information which allows including display elements like menus, forms, and choices to collect information precisely from the users

Comment (deepwired) [reply bot-builder/creating-bot.rst]: Make structured and Not structed as 2 points

File: bot-builder/creating-bot.rst [hunk 2]
```diff
@@ -1,2 +1,80 @@
...
+For instance, a customer buying a product is prompted to fill an order form. Similarly, a buyer ordering item at a
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: an item

File: bot-builder/creating-bot.rst [hunk 3]
```diff
@@ -1,2 +1,80 @@
...
+restaurant chooses the item from a list. The unstructured conversation flow includes freestyle plain text. Such as
+conversations with family, colleagues, friends and other acquaintances fall into this segment.
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: use example as "users is asking about the product", the bot responds with appropriate content

File: bot-builder/creating-bot.rst [hunk 4]
```diff
@@ -1,2 +1,80 @@
...
+
+While charting the scripting course, the designer comes across conversation of the type given below:
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: check

Comment (deepwired) [reply bot-builder/creating-bot.rst]: We will consider the following Use case...
.
Find below the scenarios ..

File: bot-builder/creating-bot.rst [hunk 5]
```diff
@@ -1,2 +1,80 @@
...
+* User wants the best hotel matching the mentioned criteria
+* User wants to make a booking
+* User gives out the stay dates
+
+Grouping conversations
+^^^^^^^^^^^^^^^^^^^^^^
+
+Before designing conversation for a chatbot, identify and understand the goals of the customer.
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: Before designing conversation flow it is vital to identify and understand the key goals the the bot has to achieve, in this case, helping a user discover and book a hotel room

File: bot-builder/creating-bot.rst [hunk 6]
```diff
@@ -1,2 +1,80 @@
...
+To be more specific, understand why the client wants to build a chatbot and what does the customer want the chatbot to do.
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: doc is for the client, who will make the bot

File: bot-builder/creating-bot.rst [hunk 7]
```diff
@@ -1,2 +1,80 @@
...
+Finding answers to this query will guide the designer to create conversations aimed at meeting end goals. For instance,
+let us take the case of a customer aspiring to build a hotel bot.
+
+We need to understand and visualize the complete flow of the chat and define the use-cases that our chatflow should cater to.
+As next steps, let’s disintegrate the whole problem into smaller chunks of solutions which will be arranged logically to form a
+whole flow for the chat.
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: remove whole

File: bot-builder/creating-bot.rst [hunk 8]
```diff
@@ -1,2 +1,80 @@
 Creating a Bot
 --------------
+
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: glossary
Story: A particular chat flow can be categaried as a story, example. Hotel Booking is a story, Hotel Information is a story
Sub Story: A minor segregation of story tasks, for example Hotel Booking would have a substory like get pricing, book room, room facilities
Node: The smallest part of our bot which will respond a message/group of messages/make an api call/or a combinition of em all/ based on what message the user sent

File: bot-builder/creating-bot.rst [hunk 9]
```diff
@@ -1,2 +1,80 @@
...
+We need to understand and visualize the complete flow of the chat and define the use-cases that our chatflow should cater to.
+As next steps, let’s disintegrate the whole problem into smaller chunks of solutions which will be arranged logically to form a
+whole flow for the chat.
+
+Stories: Discovery, Booking
+
+Sub-stories: Collect criteria for hotel match, Stay dates for booking
+
+Nodes:  Input criteria, Filter hotels based on criteria, Input stay dates, Send stay dates
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: nodes ka image

File: bot-builder/creating-bot.rst (hunk 8 above)
Comment (deepwired) [reply bot-builder/creating-bot.rst]: Move How to create below the definition... and example below the how to create

File: bot-builder/creating-bot.rst [hunk 10]
```diff
@@ -1,2 +1,80 @@
...
+
+Post creation of a sub-story, the sub-story will be displayed as depicted below.
+
+.. image:: post_sub_story_creation.png
+
+2. Creating a node
+""""""""""""""""""
+
+Enter the name of the node in the 'Name' field. Enter the system name in the 'System Name' field.
+The system name is used by the system to uniquely identify the node. The status field can be made inactive if it needs be to disabled.
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: _ and capitol

Comment (deepwired) [reply bot-builder/creating-bot.rst]: jatin

File: bot-builder/creating-bot.rst [hunk 11]
```diff
@@ -1,2 +1,80 @@
...
+The 'Start Node' checkbox can be enabled if the node if the node starts a conversation. Similarly the 'End Node' checkbox can be enabled
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: typo if the node if the node

File: bot-builder/creating-bot.rst [hunk 12]
```diff
@@ -1,2 +1,80 @@
...
+if the node if the node ends a conversation. Click on the 'Save' button once done.
```
Comment (deepwired) [inline bot-builder/creating-bot.rst]: typo if the node if the node

Comment (deepwired) [reply bot-builder/creating-bot.rst]: When an end node is reached, and that node does not have further connecting nodes, internally we   make this conversation as complete which helps indicating that a task was completed, (Visible on dashboard)

File: bot-builder/creating-bot.rst [hunk 13]
```diff
@@ -1,2 +1,80 @@
...
+Before designing conversation for a chatbot, identify and understand the goals of the customer.
+To be more specific, understand why the client wants to build a chatbot and what does the customer want the chatbot to do.
+Finding answers to this query will guide the designer to create conversations aimed at meeting end goals. For instance,
+let us take the case of a customer aspiring to build a hotel bot.
+
+We need to understand and visualize the complete flow of the chat and define the use-cases that our chatflow should cater to.
+As next steps, let’s disintegrate the whole problem into smaller chunks of solutions which will be arranged logically to form a
+whole flow for the chat.
+
+Stories: Discovery, Booking
```
Comment (superprat) [inline bot-builder/creating-bot.rst]: add a note stating the fact that nodes cannot be connected across stories, stories are completely independent

File: bot-builder/creating-bot.rst [hunk 14]
```diff
@@ -1,2 +1,80 @@
...
+
+Sub-stories: Collect criteria for hotel match, Stay dates for booking
```
Comment (superprat) [inline bot-builder/creating-bot.rst]: Add a note that that linking can happen here. across substories

File: bot-builder/creating-bot.rst (hunk 9 above)
Comment (superprat) [inline bot-builder/creating-bot.rst]: A line or two, based on when to make something 1 node or multiple nodes.

**Pull request merged**